packages/core/src/modules/employment/queries/list-job-offers.ts

```diff
@@ -0,0 +1,73 @@
+import { ListSearchParams } from '@/shared/types';
+import { Company, EmploymentType, JobOfferStatus } from '../employment.types';
+import { db } from '@/infrastructure/database';
+import { sql } from 'kysely';
+import { z } from 'zod';
+import { ExtractValue } from '@oyster/types';
+
+const JobOfferFilterParams = z.object({
+  company: Company.shape.id.nullable().catch(null),
+  employmentType: z.nativeEnum(EmploymentType),
+  location: z.string().trim().min(1).nullable().catch(null),
+  status: z.nativeEnum(JobOfferStatus),
+});
+
+const JobOfferFilterParam = JobOfferFilterParams.keyof().enum;
+
+const JobOfferSearchParams = ListSearchParams.omit({
+  timezone: true,
+  search: true,
+}).merge(JobOfferFilterParams);
+
+type JobOfferFilterParam = ExtractValue<typeof JobOfferFilterParam>;
+type JobOfferSearchParams = z.infer<typeof JobOfferSearchParams>;
+
+async function listJobOffers({
+  company,
+  employmentType,
+  location,
+  status,
+}: JobOfferSearchParams) {
+  const query = db
+    .selectFrom('jobOffers')
+    .$if(!!company, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb
+            .selectFrom('jobOffers')
+            .where('companies.crunchbaseId', '=', company)
+        );
+      });
+    })
+    .$if(!!employmentType, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb
+            .selectFom('jobOffers')
+            .where('jobOffers.employmentType', '=', employmentType)
+        );
+      });
+    })
+    .$if(!!location, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb.selectFrom('jobOffers').where('jobOffers.location', '=', location)
+        );
+      });
+    })
```

Comment:
The `location` that's passed in is going to be some raw text string (ie: Los Angeles, CA, USA) so the query as is will only return job offers that match that string exactly. I'm thinking we should return job offers with locations that are nearby (say within 25 miles).

Instead of passing in the `location`, maybe we can pass in the latitude/longitude and do something like this:

```ts
 .$if(!!locationLatitude && !!locationLongitude, (query) => {
    return query.where(
      sql`location_coordinates <@> point(${locationLongitude}, ${locationLatitude})`,
      '<=',
      25
    );
  })
```

For reference, the `<@>` operator computes the distance between 2 geographical points and returns the distance in miles! See [here](https://www.postgresql.org/docs/current/earthdistance.html#EARTHDISTANCE-POINT-BASED) for documentation.